rest/admin_api_test.go

```diff
@@ -3370,3 +3370,47 @@ func TestIncludeRuntimeStartupConfig(t *testing.T) {
 	assert.Equal(t, db.ActiveReplicatorTypePushAndPull, replCfg.Direction)
 
 }
+
+func TestPersistentConfigConcurrency(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("This test only works against Couchbase Server")
+	}
+
+	defer base.SetUpTestLogging(base.LevelInfo, base.KeyHTTP)()
+
+	// Start SG with no databases in bucket(s)
+	config := bootstrapStartupConfigForTest(t)
+	sc, err := setupServerContext(&config, true)
+	require.NoError(t, err)
+	serverErr := make(chan error, 0)
+	go func() {
+		serverErr <- startServer(&config, sc)
+	}()
+	require.NoError(t, sc.waitForRESTAPIs(time.Second*5))
+
+	// Get a test bucket, and use it to create the database.
+	tb := base.GetTestBucket(t)
+	defer func() {
+		fmt.Println("closing test bucket")
+		tb.Close()
+	}()
+	resp := adminRequest(t, http.MethodPut, "/db/",
+		`{"bucket": "`+tb.GetName()+`", "num_index_replicas": 0}`,
+	)
+	assert.Equal(t, http.StatusCreated, resp.StatusCode)
+
+	// Get config
+	resp = adminRequest(t, "GET", "/db/_config?redact=false", "")
+	assert.Equal(t, http.StatusOK, resp.StatusCode)
+
+	eTag := resp.Header.Get("ETag")
```

Comment:
Might want to check this, as I'm not sure the assertion below will fail if etag is an empty string (it will just always do an unsafe PUT)

```suggestion
	eTag := resp.Header.Get("ETag")
	assert.NotEqual(t, "", eTag)
```

Reply:
Good point